packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L1ETHGateway.sol

```diff
@@ -68,66 +67,100 @@ contract OVM_L1ETHGateway is iOVM_L1ETHGateway, OVM_CrossDomainEnabled, Lib_Addr
+     *        solely as a convenience for external contracts. Aside from enforcing a maximum
+     *        length, these contracts provide no guarantees about its content.
      */
-    function deposit()
+    function deposit(
+        uint32 _l2Gas,
+        bytes calldata _data
+    )
         external
         override
         payable
     {
-        _initiateDeposit(msg.sender, msg.sender);
+        _initiateDeposit(
+            msg.sender,
+            msg.sender,
+            _l2Gas,
+            _data
+        );
     }
 
     /**
-     * @dev deposit an amount of ETH to a recipients's balance on L2
-     * @param _to L2 address to credit the withdrawal to
+     * @dev Deposit an amount of ETH to a recipient's balance on L2.
+     * @param _to L2 address to credit the withdrawal to.
+     * @param _l2Gas Gas limit required to complete the deposit on L2.
+     * @param _data Optional data to forward to L2. This data is provided
+     *        solely as a convenience for external contracts. Aside from enforcing a maximum
+     *        length, these contracts provide no guarantees about its content.
      */
     function depositTo(
-        address _to
+        address _to,
+        uint32 _l2Gas,
+        bytes calldata _data
     )
         external
         override
         payable
     {
-        _initiateDeposit(msg.sender, _to);
+        _initiateDeposit(
+            msg.sender,
+            _to,
+            _l2Gas,
+            _data
+        );
     }
 
     /**
      * @dev Performs the logic for deposits by storing the ETH and informing the L2 ETH Gateway of the deposit.
-     *
-     * @param _from Account to pull the deposit from on L1
-     * @param _to Account to give the deposit to on L2
+     * @param _from Account to pull the deposit from on L1.
+     * @param _to Account to give the deposit to on L2.
+     * @param _l2Gas Gas limit required to complete the deposit on L2.
+     * @param _data Optional data to forward to L2. This data is provided
+     *        solely as a convenience for external contracts. Aside from enforcing a maximum
+     *        length, these contracts provide no guarantees about its content.
      */
     function _initiateDeposit(
         address _from,
-        address _to
+        address _to,
+        uint32 _l2Gas,
+        bytes memory _data
     )
         internal
     {
         // Construct calldata for l2ETHGateway.finalizeDeposit(_to, _amount)
-        bytes memory data =
+        bytes memory message =
             abi.encodeWithSelector(
                 iOVM_L2DepositedToken.finalizeDeposit.selector,
+                _from,
                 _to,
-                msg.value
+                msg.value,
+                _data
             );
 
         // Send calldata into L2
         sendCrossDomainMessage(
             ovmEth,
-            data,
-            getFinalizeDepositL2Gas
+            message,
+            _l2Gas
```

Comment:
For consistency, can we also switch the bytes <> uint arguments for message and l2Gas here?

Reply:
That would be _less_ consistent with the interface of the L1CrossDomainMessenger, which IMO would look odd here: 

```
    function sendCrossDomainMessage(
        address _crossDomainTarget,
        bytes memory _message,
        uint32 _gasLimit
    )
        internal
    {
        getCrossDomainMessenger().sendMessage(_crossDomainTarget, _message, _gasLimit);
    }
}
```

Reply:
Despite the comment above, I decided to just go ahead and make the change. I agree we should aim to stick with this convention moving forward.